domains/transfer-agreements/TransferAgreements.API/API.IntegrationTests/Factories/TransferAgreementsApiWebApplicationFactory.cs

```diff
@@ -1,16 +1,58 @@
 using System;
+using System.Data.Common;
 using System.IdentityModel.Tokens.Jwt;
 using System.Net.Http;
 using System.Net.Http.Headers;
 using System.Security.Claims;
 using System.Text;
+using System.Threading.Tasks;
+using API.Data;
+using API.Options;
+using Microsoft.AspNetCore.Hosting;
 using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
 using Microsoft.IdentityModel.Tokens;
+using Testcontainers.PostgreSql;
+using Xunit;
 
 namespace API.IntegrationTests.Factories;
 
-public class TransferAgreementsApiWebApplicationFactory : WebApplicationFactory<Program>
+public class TransferAgreementsApiWebApplicationFactory : WebApplicationFactory<Program>, IAsyncLifetime
 {
+    private readonly PostgreSqlContainer testContainer = new PostgreSqlBuilder().WithImage("postgres:15.2").Build();
```

Comment:
It is the version currently being used by the cnpg operator. There is no pinning of the postgres version used in k8s, so it follows the default for the cnpg operator. 
I chose to explicitly pin it in the integration test + Docker compose setup, so that we at least here runs with the same version everywhere.